roles/StackStorm.ewc_smoketests/tasks/main.yml

```diff
@@ -1,89 +0,0 @@
----
-
-# Small suite of smoke tests to ensure that EWC role has deployed as expected
-
-- name: auth | Add a new ewc_smoke_tests_user in st2 htpasswd file
-  become: true
-  htpasswd:
-    path: /etc/st2/htpasswd
-    name: "{{ ewc_smoke_tests_user }}"
-    password: "{{ ewc_smoke_tests_password }}"
-  changed_when: no
-  notify:
-    - restart st2api
-    - restart st2stream
-  tags:
-    - ewc-smoke-tests
-
-- name: Copy smoke tests RBAC roles to /opt/stackstorm/rbac/roles directory
-  become: yes
-  template:
-    src: rbac_roles/roles.yml.j2
-    dest: /opt/stackstorm/rbac/roles/{{ item.name }}.yaml
-    owner: st2
-    group: st2
-  loop: "{{ ewc_smoke_tests_rbac.roles }}"
-  changed_when: no
-  when: ewc_smoke_tests_rbac.roles is defined
-  notify:
-    - reload ewc_rbac
-  tags:
-    - ewc-smoke-tests
-
-- name: Copy smoke tests RBAC assignments to /opt/stackstorm/rbac/assignments directory
-  become: yes
-  template:
-    src: rbac_assignments/assignments.yml.j2
-    dest: /opt/stackstorm/rbac/assignments/{{ item.name }}.yaml
-    owner: st2
-    group: st2
-  loop: "{{ ewc_smoke_tests_rbac.assignments }}"
-  changed_when: no
-  when: ewc_smoke_tests_rbac.assignments is defined
-  notify:
-    - reload ewc_rbac
-  tags:
-    - ewc-smoke-tests
-
-- meta: flush_handlers
-  tags:
-    - ewc-smoke-tests
-
-- name: Get authentication token for ewc_smoke_tests  # Note this will not use LDAP.
-  command: st2 auth "{{ ewc_smoke_tests_user }}" -p "{{ ewc_smoke_tests_password }}"  -t
-  register: st2_token_smoke_tests_user
-  changed_when: no
-  tags:
-    - ewc-smoke-tests
-
-- name: Test a simple core.local action as user ``ewc_smoke_tests_user``
-  command: st2 run core.local -- date -R
-  environment:
-    ST2_AUTH_TOKEN: "{{ st2_token_smoke_tests_user.stdout }}"
-  changed_when: no
-  tags:
-    - ewc-smoke-tests
-
-- name: Test some other action that "{{ ewc_smoke_tests_user }}" cannot run
-  command: st2 run core.http url="https://www.google.com"
-  environment:
-    ST2_AUTH_TOKEN: "{{ st2_token_smoke_tests_user.stdout }}"
-  ignore_errors: yes
-  changed_when: no
-  register: ewc_smoke_tests_forbidden_action
-  tags:
-    - ewc-smoke-tests
-
-- name: Assert forbidden error was indeed thrown
```

Comment:
Looking at this file, EWC_smoketests are not bad at all.
Could we on-board them for the Community version to verify that LDAP/RBAC configuration really works?

Reply:
Oh, good point. Will do that until end of this week.

Reply:
I'm almost done. Just need to restructure the code a bit due to a change in the recent ansible versions that prevents the flush_handlers task to be executed. Solution will be ready tomorrow.

Reply:
Ok, a warning led me into the wrong direction and the issue was actually something else. 

@armab I added the tests again but I had to remove the `changed_when: no` when deploying the roles and assignment template for the smoke test. Otherwise the `reload rbac` would not be triggered. I don't see an issue with it. Do you think it's fine, too? Otherwise I'll remove the notify statements and execute the handler as task regardless of any changes on these files.

Reply:
Good to me if that works. I remember it might be needed for the Ansible idempotence checks, but don't remember for sure. As commented below let's enable the tests in CI and see the actual behavior.